src/Engine/Scene/Parsing/SceneLoader.cpp

```
@@ -462,69 +543,84 @@ void SceneLoader::processScene(const YAML::Node &rootNode, std::string &currentE
+            // ----- ENTITY IS CUSTOM-DEFINED => LOAD SPECIFIED COMPONENTS -----
+        std::vector<YAMLParseCtx> components{};
+        for (YAML::Node componentNode : entityNode[YAMLScene::Entity_Components]) {
+            ComponentName componentType = componentNode[YAMLScene::Entity_Components_Type].as<ComponentName>();
+            
+            parseContext.selfComponents = &componentNodes[entityName];
+            parseContext.currentComponentType = componentType;
 
-            m_serialRegistry.deserialize(YAMLScene::Body_Prefix, static_cast<IParseCtx *>(&parseContext));
+            componentNodes[entityName][componentType] = componentNode;
+            entityParseContexts.emplace_back(parseContext);
         }
```

Cppcheck (2.20.0)

[performance] Function 'getRenderSpaceEntity()' should return member 'm_renderSpace' by const reference.

(returnByReference)